src/http/matcher/mod.rs

```diff
@@ -602,6 +692,7 @@ impl<State, Body> crate::service::Matcher<State, Request<Body>> for HttpMatcherK
             HttpMatcherKind::Header(header) => header.matches(ext, ctx, req),
             HttpMatcherKind::Socket(socket) => socket.matches(ext, ctx, req),
             HttpMatcherKind::Any(all) => all.iter().matches_or(ext, ctx, req),
+            HttpMatcherKind::Custom(f) => f.matches(ext, ctx, req),
```

Comment:
you call it `f` as you assume it will always be a function, but that's a wrong assumption. It can be used to implemented any kind of matcher, while still being able to plug-in to the enum http enum matcher.

In that spirit maybe call it `matcher` instead of `f`?

Reply:
That makes more sense, agreed.